core/common/validator-core/src/main/java/org/eclipse/edc/validator/dataaddress/KafkaDataAddressValidator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.validator.dataaddress;
+
+import org.eclipse.edc.spi.types.domain.DataAddress;
+import org.eclipse.edc.validator.spi.ValidationResult;
+import org.eclipse.edc.validator.spi.Validator;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.eclipse.edc.spi.dataaddress.KafkaDataAddressSchema.BOOTSTRAP_SERVERS;
+import static org.eclipse.edc.spi.dataaddress.KafkaDataAddressSchema.TOPIC;
+import static org.eclipse.edc.validator.spi.Violation.violation;
+
+public class KafkaDataAddressValidator implements Validator<DataAddress> {
```

Comment:
I can agree with this, I wanted to understand how deep to go with the refactoring.

Likely, every data type should have its own `Schema` interface (for now they are both in the `core-spi` modules) and a `Validator` implementation (that currently I put in the `validator-core` module).

Splitting this vertically, we could end up having, for every `DataAddress` type, an `spi` and a `validator` extension, like:
`dataaddress-httpdata-spi` - `validator-dataaddress-httpdata`
and
`dataaddress-kafka-spi` - `validator-dataaddress-kafka`

with the `spi` containing only the `Schema` interface and the `validator` containing the `Validator` implementation and the registration extension.

wdyt @jimmarino ?

Reply:
That works for me. One relly minor nit would be to use "data-address" as opposed to "dataadress" since it is easier to read, but that is only a minor suggestion.